### ISSUE TITLE
BUG: processing overlapping data frame

### ISSUE THREAD

**Issue by cang-mang** [litespeedtech/lsquic#106]
We are using lsquic-client-1.21.2 as client, and chrome-quic as server. In below case we find the stream dies.

First:
The server sends: [2346,3518) [3518,4096)

Second:
The client reads all, and set the read_offset to 4096. However from server side, the ACK [3518,4096) is lost due to network condition.

Third:
The server retransfers: [3518,4690) [4690,5862) ...

Last:
The client can read nothing any more. Because the first frame's offset is 3518, but the stream's read_offset is 4096, and the code says:
if (frame && frame->data_frame.df_offset + frame->data_frame.df_read_off == read_offset)

The stream will never hit this statement.

The newest version of code still has this BUG.

**Comment by dtikhonov** [comment]
Thank you for the bug report!  It is only partially correct: the bug exists in 1.21.2, but does not exist on the master branch.  To prove it, I [added a unit test](https://github.com/litespeedtech/lsquic/commit/35ac25bb73afa79b03d51bbcb6ac866c9eaf996d) that runs through the scenario you described.

Interestingly, the bug in 1.21.2 was [fixed in 2.2.0](https://github.com/litespeedtech/lsquic/commit/5392f7a3b05e515b403e314aa8f0a40d91135d5d#diff-fa3c338b42c0479021484be57be6dcadR279):

```c
        else if (read_offset > DF_OFF(new_frame))
            return INS_FRAME_OVERLAP                            | CASE('N');
```

Why do you using the 1.x version of the library instead of the 2.x?

If you'd like, we can release 1.21.3 with this bug fix.

**Comment by litespeedtech** [comment]
Closing -- not a bug.
